### ISSUE TITLE
Not able to disable body transformers when producing jimple

### ISSUE THREAD

**Issue by linghuiluo** [soot-oss/soot#1545]
I noticed that there are a few body transformers are used in `PackManager.java` without checking if the options are disabled when producing jimple (see below). This is inconsistent to the normal usage in JimpleBodyPack through `PackManager.v().getTransform("jb.cp").apply(b);`

The question is, is this intentionally? I came across this, since sometimes you might want to disable the transformers, e.g. variables that are only used once won't be optimized. 
 
https://github.com/soot-oss/soot/blob/develop/src/main/java/soot/PackManager.java#L1000

```
  if (produceJimple) {
        Body body = m.retrieveActiveBody();
        // Change
        CopyPropagator.v().transform(body);
        ConditionalBranchFolder.v().transform(body);
        UnreachableCodeEliminator.v().transform(body);
        DeadAssignmentEliminator.v().transform(body);
        UnusedLocalEliminator.v().transform(body);
```
@anddann @mbenz89  any opinion on this?

**Comment by patricklam** [comment]
Yes, that is intentional. Without them the produced code would be quite
bloated due to the transformation from bytecode to three address code and
back again. You can try to comment them out and inspect the generated code.

**Comment by linghuiluo** [comment]
@patricklam Thank you for your answer! I know it is for optimization purpose. However, I feel this way of using the transformers makes the soot phase options useless if people don't want to modify the code by themselves. Yesterday a colleague of mine asked if he could turn off some jimple body transformers such as the CopyPropagator because he wants to perform constant propagation for students, but soot optimizes the local variables.  I pointed him to use `Options.v().setPhaseOption("jb.cp", "enabled:false");`. However, this didn't help due to the direct usage of the transformers as I pointed out above.
